src/test/java/org/elasticsearch/index/mapper/date/SimpleDateMappingTests.java

```diff
@@ -439,4 +427,21 @@ public void testNumericResolution() throws Exception {
                 .bytes());
         assertThat(getDateAsMillis(doc.rootDoc(), "date_field"), equalTo(44000L));
     }
+
+    public void testThatOlderIndicesAllowNonStrictDates() throws Exception {
+        String mapping = XContentFactory.jsonBuilder().startObject().startObject("type")
+                .startObject("properties").startObject("date_field").field("type", "date").endObject().endObject()
+                .endObject().endObject().string();
+
+        String indexName = "test-" + randomAsciiOfLength(10).toLowerCase(Locale.ROOT);
+        IndexService index = createIndex(indexName, settingsBuilder().put(IndexMetaData.SETTING_VERSION_CREATED, Version.V_1_5_2_ID).build());
+        client().admin().indices().preparePutMapping(indexName).setType("type").setSource(mapping).get();
+        DocumentMapper defaultMapper = index.mapperService().documentMapper("type");
+
+        defaultMapper.parse("type", "1", XContentFactory.jsonBuilder()
+                .startObject()
+                .field("date_field", "1-1-1T00:00:44.000Z")
+                .endObject()
+                .bytes());
+    }
```

Comment:
can we also check we can parse a proper full formatted date here? I always like to test both "good" and "bas"